x-pack/plugins/infra/public/pages/metrics/hosts/components/chart/lens_wrapper.tsx

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useEffect, useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+
+import { Action } from '@kbn/ui-actions-plugin/public';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { BrushTriggerEvent } from '@kbn/charts-plugin/public';
+import { EuiFlexGroup } from '@elastic/eui';
+import { EuiFlexItem } from '@elastic/eui';
+import { EuiLoadingChart } from '@elastic/eui';
+import { Filter, Query, TimeRange } from '@kbn/es-query';
+import { useIntersectedOnce } from '../../../../../hooks/use_intersection_once';
+import { InfraClientSetupDeps } from '../../../../../types';
+import { LensAttributes } from '../../../../../common/visualizations';
+
+export interface Props {
+  id: string;
+  attributes: LensAttributes | null;
+  dateRange: TimeRange;
+  query: Query;
+  filters: Filter[];
+  extraActions: Action[];
+  lastReloadRequestTime?: number;
+  style?: React.CSSProperties;
+  onBrushEnd?: (data: BrushTriggerEvent['data']) => void;
+}
+
+export const LensWrapper = ({
+  attributes,
+  dateRange,
+  filters,
+  id,
+  query,
+  extraActions,
+  style,
+  onBrushEnd,
+  lastReloadRequestTime,
+}: Props) => {
+  const intersectionRef = React.useRef(null);
+
+  const [currentLastReloadRequestTime, setCurrentLastReloadRequestTime] = useState<
+    number | undefined
+  >(lastReloadRequestTime);
+  const {
+    services: { lens },
+  } = useKibana<InfraClientSetupDeps>();
+  const { intersectedOnce, intersection } = useIntersectedOnce(intersectionRef, {
+    threshold: 1,
+  });
+
+  const EmbeddableComponent = lens.EmbeddableComponent;
+
+  useEffect(() => {
+    if ((intersection?.intersectionRatio ?? 0) === 1) {
+      setCurrentLastReloadRequestTime(lastReloadRequestTime);
+    }
+  }, [intersection?.intersectionRatio, lastReloadRequestTime]);
+
+  const isReady = attributes && intersectedOnce;
```

Comment:
This will make the chart **only** load if it becomes visible in the viewport. There are many simultaneous requests to `bsearch` and I've experienced some performance degradation because of that. 

This doesn't fully solve the problem as the user can scroll down and load all the charts simultaneously. Another downside is that the user has to wait an additional time to load the charts at the bottom of the page. This is more noticeable when trying to load the page using a wider date range.